src/pages/Article.js

```diff
@@ -1,20 +1,34 @@
-import { Avatar, Card, CardHeader } from '@mui/material';
+import { Avatar, Card, CardHeader, CircularProgress } from '@mui/material';
 import React, { useEffect, useState } from 'react';
 import { Link, useParams } from 'react-router-dom';
 import { getSingleArticle } from '../helpers/axiosConfig';
 import HomeIcon from '@mui/icons-material/Home';
-
+import { getSingleArticleRedux } from '../features/articleSlice';
 import './Article.css';
+import { useDispatch, useSelector } from 'react-redux';
+import { getArticles } from '../features/selectors';
 
 const Article = () => {
   let { id } = useParams();
+  const dispatch = useDispatch();
+  const { article, loadingSingleArticle } = useSelector(getArticles);
   const [articleData, setArticleData] = useState();
   useEffect(() => {
+    dispatch(getSingleArticleRedux(id));
     getSingleArticle(id).then(({ data }) => {
       setArticleData(data);
     });
   }, [id]);
 
+  if (typeof articleData !== 'object') {
+    return (
+      <div className="empty-array">
+        <h2>We are looking for article </h2>
+        {loadingSingleArticle && <CircularProgress />}
+      </div>
+    );
+  }
```

Comment:
You don't need `articleData` state here since you've got `article` from selector. Also, at the moment you're fetching article twice (first time with redux `getSingleArticleRedux` action, and the second time with `getSingleArticle` function)
```suggestion
  useEffect(() => {
    dispatch(getSingleArticleRedux(id));
  }, [id]);

  if (loadingSingleArticle) {
    return (
      <div className="empty-array">
        <h2>We are looking for article </h2>
        <CircularProgress />
      </div>
    );
  }
```

Reply:
I repaired double fetching data and remove unused state